src/table/table-role/grid-navigation.tsx

```diff
@@ -74,8 +74,9 @@ class GridNavigationProcessor {
 
   // State
   private focusedCell: null | FocusedCell = null;
-  private focusables = new Map<Element, boolean>();
+  private focusables = new Set<Element>();
   private focusHandlers = new Map<Element, FocusableChangeHandler>();
+  private focusablesState = new WeakMap<Element, boolean>();
```

Comment:
What is the advantage of having `focusables` and `focusablesState` separated?

Reply:
The entries from `focusablesState` are not removed upon unregistering. This way, if the same DOM node is unregistered and registered again (happens on every element render), certain extra work can be avoided because the focusable state remains cached.